### PR TITLE
Fix/details query and testing

### PR DESCRIPTION
**Pull request by rdtho2525**
# Selector FE PR

## What does this merge do? Check all that apply.
- [ ] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [X] Cleans up (logs, tests, etc.)

## Summary of changes
- Conducts E2E testing for DetailsModal component
- Refactors `Details.css`

## Notes for reviewer (any known bugs, areas of interest, etc.)
- Run `npm run cypress`
- Select `details-test.js`
- All tests should be passing

## Next Steps (Future Iterations)
- Setup CI/CD for entire project

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (monshan) [inline cypress/fixtures/example.json:5]: Might be able to delete this example file

File: cypress/fixtures/details-fixture.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[
```
Comment (Max9545) [inline cypress/fixtures/details-fixture.json]: {data: album {info}}

File: cypress/integration/details-test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+describe('', () => {
```
Comment (monshan) [inline cypress/integration/details-test.js:1]: Maybe we can name it something like 'Modal' or 'User can observe an accurate modal'

File: src/components/Details/Details.js [hunk 4]
```diff
@@ -29,40 +29,39 @@ const ALBUMS = gql`
...
-
-  console.log(data)
   
   return (
     <div className="modal-container">
-      <section className="modal">
+      <section className="modal" data-cy="modal">
         <article className="box left">
-          <img className="album-cover shadow" src={data.album.coverImage} alt={`${data.album.title} album cover`}/>
+          <img data-cy="album-cover" className="album-cover shadow" src={data.album.coverImage} alt={`${data.album.title} album cover`}/>
```
Comment (Max9545) [inline src/components/Details/Details.js:43]: lets make sure the cars and modal classes don't interfere with each other once all merged

File: cypress/fixtures/details-fixture.json (hunk 2 above)
Comment (rdtho2525) [reply cypress/fixtures/details-fixture.json]: The following commit address fixes this configuration issue:
"Reconfigure fixture object" 5df2c62

File: cypress/integration/details-test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  it('should allow users to add the current album to their collection of favorites', () => {
+    cy.get('[data-cy=favorites-button')
+      .should('exist')
+  });
+
+  it('should allow users to visit discogs.com from the modal', () => {
+    cy.get('[data-cy=discogs-link')
+      .should('exist')
+  });
```
Comment (monshan) [inline cypress/integration/details-test.js:25]: Might be just these two that will get expanded on later, I'm tempted to suggest just the re-naming of these tests to 'render xxx' or 'xxx exist'

File: cypress/integration/details-test.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  it('should display a playlist where users can preview that album', () => {
+    cy.get('[data-cy=web-player')
+      .should('have.attr', 'src')
+      .should('include', 'https://open.spotify.com/embed/album/')
+  });
+
+  it('should allow users to close the modal window', () => {
+    cy.get('[data-cy=close-button')
+      .should('exist')
```
Comment (monshan) [inline cypress/integration/details-test.js:50]: Same here with the elements above

**Review by monshan — APPROVED** [review]
Great work getting the testing done for the modal to check for existence and visibility

1. Reame some tests defined above
2. Rename data-cy properties to have the modal prefix

File: cypress/fixtures/example.json (hunk 1 above)
Comment (rdtho2525) [reply cypress/fixtures/example.json:5]: Noted: will delete in the next branch.

File: cypress/integration/details-test.js (hunk 3 above)
Comment (rdtho2525) [reply cypress/integration/details-test.js:1]: Thank you for catching this oversight. Your suggestion will be written in the next branch.

File: cypress/integration/details-test.js (hunk 5 above)
Comment (rdtho2525) [reply cypress/integration/details-test.js:25]: As you alluded to, the intent was/is to expand on these later once functionality is added to these elements.  Whether those behavioral tests replace the above or just supplement the existence tests, I'm content.

File: cypress/integration/details-test.js (hunk 6 above)
Comment (rdtho2525) [reply cypress/integration/details-test.js:50]: Same as above.

**Pull request merged**